Added information about deployer 2.0 (#8005)
magento/devdocs: src/contributor-guide/contributing.md

#### Magento Edition:

Append the following text to your PR comment to specify the Magento Commerce edition to use when you [Deploy a vanilla Magento instance](#vanilla-pr) or [Deploy an instance based on PR changes](#deploy-pr).

```text
with edition {$edition}
```

For`edition`, use either of the following values:

-  `ee` deploys the Magento Commerce edition
-  `b2b` deploys Magento Commerce with B2B modules.

#### Environment:

Append the following text to your PR comment to specify the version for applications and services to use when you [Deploy a vanilla Magento instance](#vanilla-pr) or [Deploy an instance based on PR changes](#deploy-pr).

```text
with env PHP {$phpVersion}, search-engine ElasticSearch {$searchEngineVersion}, database {$dbEngine} {$dbEngineVersion}
```

Replace variables in the command with the following values as needed for your environment:

-  `phpVersion`–Specify the PHP version for the instance.

-  `searchEngineVersion`–Specify the Elasticsearch version for the instance.

-  `dbEngine`–Specify the database type, either `MariaDB` or `MySQL`.

-  `dbEngineVersion`–Specify the version of the database engine for the instance.

> For example, append the following text to the PR comment to deploy an instance with PHP 7.4, Elasticsearch version 7, and MariaDB version 10.4.

Author: ```suggestion

For example, append the following text to the PR comment to deploy an instance with PHP 7.4, Elasticsearch version 7, and MariaDB version 10.4.
```

-  Magento EngCom Team
-  [Contributor]({{ site.baseurl }}/contributor-guide/contributors.html)

### Customize deployed instances {#customize-deployed-instances}

> In some cases a custom environment is necessary in order to test an issue, or a pull request, to achieve such a need there are a set of commands that, combined with the [Deploy vanilla Magento instance](#vanilla-pr) or [Deploy instance based on PR changes](#deploy-pr), will allow you to customize your deployed instance environment.

Reviewer: ```suggestion
In some cases a custom environment is required to test an issue or a pull request. You can create a custom environment by appending custom configuration settings to the PR comment to [Deploy vanilla Magento instance](#vanilla-pr) or [Deploy instance based on PR changes](#deploy-pr).
```

#### Magento Edition:

> By appending the text below to [Deploy vanilla Magento instance](#vanilla-pr) or [Deploy instance based on PR changes](#deploy-pr) commands you will be able to specify which edition you would like to have deployed.

Reviewer: ```suggestion
You can append the following text to your PR comment to specify the Magento Commerce edition to use when you [Deploy a vanilla Magento instance](#vanilla-pr) or [Deploy an instance based on PR changes](#deploy-pr).
```

```text
with edition {$edition}
```

> For `edition`, the supported values are `ee` to deploy Magento Commerce edition or `b2b` to deploy the Magento Commerce with B2B modules.

Author: The Magento Open Source is the default, this command is as an addition to the other 2 commands, so if you don't add this by default you get the Magento Open Source.

> For `edition`, the supported values are `ee` to deploy Magento Commerce edition or `b2b` to deploy the Magento Commerce with B2B modules.

Reviewer: ```suggestion
For`edition`, use either of the following values:

-  `ee` deploys the Magento Commerce edition
-  `b2b` deploys Magento Commerce with B2B modules.
```

#### Environment:

> By appending the text below to [Deploy vanilla Magento instance](#vanilla-pr) or [Deploy instance based on PR changes](#deploy-pr) commands you will be able to specify different versions of the applications/services running on the instance.

Reviewer: ```suggestion
You can append the following text to your PR comment to specify the version for applications and services to use when you [Deploy a vanilla Magento instance](#vanilla-pr) or [Deploy an instance based on PR changes](#deploy-pr).
```

```text

> with env[ironment] PHP {$phpVersion}, search-engine ElasticSearch {$searchEngineVersion}, database {$dbEngine} {$dbEngineVersion}

Reviewer: should this be:

with env PHP {$phpVersion}, ....

Author: In this case it can be either:

`with env` or `with environment`

Maybe we should just leave the short form of it.

> with env[ironment] PHP {$phpVersion}, search-engine ElasticSearch {$searchEngineVersion}, database {$dbEngine} {$dbEngineVersion}

Author: ```suggestion
with env PHP {$phpVersion}, search-engine ElasticSearch {$searchEngineVersion}, database {$dbEngine} {$dbEngineVersion}
```

```

> You can replace the variables above with the following:
> 
> -  For `phpVersion`, you can set the PHP version that you want the instance to run on.
> 
> -  For `searchEngineVersion`, you can set the Elasticsearch version that you want the instance to run.
> 
> -  For `dbEngine`, you can choose between `MariaDB` or `MySQL`.
> 
> -  For `dbEngineVersion`, you can replace with the version of the database engine that you chose.
> 

Reviewer: ```suggestion
Replace variables in the command with the following values as needed for your environment:

-  `phpVersion`–Specify the PHP version for the instance.

-  `searchEngineVersion`–Specify the Elasticsearch version for the instance.

-  `dbEngine`–Specify the database type, either `MariaDB` or `MySQL`.

-  `dbEngineVersion`–Specify the version of the database engine for the instance.
```

> As an example you can have an environment defined as shown below:

Reviewer: ```suggestion
For example, append the following text to the PR comment to deploy an instance with PHP 7.4, Elasticsearch version 7, and MariaDB version 10.4.
```

```text

> with env PHP 7.4, search-engine ElasticSearch 7, database MariaDB 10.4

Reviewer 2: ```suggestion
with env PHP 7.4, search-engine Elasticsearch 7, database MariaDB 10.4
```

```

> Please be aware such feature should be only used in specific cases, and the general recommendation is to not change the default environment.

Reviewer 2: ```suggestion
Please be aware, these features should be only used in specific cases  and the general recommendation is not to change the default environment.
```

> Please be aware, these features should be only used in specific cases  and the general recommendation is not to change the default environment.

Reviewer: ```suggestion
{:.bs-callout-info}
We generally recommend that you deploy the default environment. Use the custom configuration options only when you require a special configuration to test specfic use cases.
```

#### Add Extensions:

> By appending the text below to [Deploy vanilla Magento instance](#vanilla-pr) or [Deploy instance based on PR changes](#deploy-pr) commands you will be able to specify extensions that you would like to add to your instance.

Reviewer: ```suggestion
You can append the following text to your PR comment to specify extensions to add to an instance when you [Deploy a vanilla Magento instance](#vanilla-pr) or [Deploy an instance based on PR changes](#deploy-pr).
```

```text
with extensions {$extensionRepo}
```

> For `extensionRepo`, you can specify extension repositories that will be additionally used when assembling your instance. You can also define the branch by using the pattern: org/repo-name:branch-name.

Reviewer 2: ```suggestion
For `extensionRepo`, you can specify extension repositories that will be additionally used when compiling your instance. You can also define the branch by using the pattern: org/repo-name:branch-name.
```

> With such a feature you can specify multiple modules by separating each module with comma. As an example:

Reviewer 2: ```suggestion
With this feature you can specify multiple modules by separating each module with comma. As an example:
```

> For `extensionRepo`, you can specify extension repositories that will be additionally used when compiling your instance. You can also define the branch by using the pattern: org/repo-name:branch-name.
> 
> With this feature you can specify multiple modules by separating each module with comma. As an example:
> 
> ```text
> with extensions magento/security-package:1.0-develop, magento/security-package-ee
> ```

Reviewer: ```suggestion
For `extensionRepo`, specify one or more extension repositories to include when compiling your instance.  If you specify multiple repositories, use a comma after each repository. You can specify a specific branch in a repository using the pattern: `org/repo-name:branch-name`.  For example:

```text
with extensions magento/security-package:1.0-develop, magento/security-package-ee
```
```

#### Remove Extensions:

> By appending the text below to [Deploy vanilla Magento instance](#vanilla-pr) or [Deploy instance based on PR changes](#deploy-pr) commands you will be able to remove extensions that you would like to be removed on your instance.

Reviewer 2: ```suggestion
By appending the text below to [Deploy vanilla Magento instance](#vanilla-pr) or [Deploy instance based on PR changes](#deploy-pr) commands you will be able to specify extensions that you would like to be removed on your instance.
```

> By appending the text below to [Deploy vanilla Magento instance](#vanilla-pr) or [Deploy instance based on PR changes](#deploy-pr) commands you will be able to specify extensions that you would like to be removed on your instance.

Reviewer: ```suggestion
Append the following text to your PR comment to specify extensions any extensions that you want to remove from the instance when you [Deploy a vanilla Magento instance](#vanilla-pr) or [Deploy an instance based on PR changes](#deploy-pr).
```

```text
without extensions {$extensionRepo}
```

> For `extensionRepo`, you can specify extension repositories that will be removed when assembling your instance.

Reviewer 2: ```suggestion
For `extensionRepo`, you can specify extension repositories that will be removed when compiling your instance.
```

> For `extensionRepo`, you can specify extension repositories that will be removed when compiling your instance.
> 
> With such a feature you can specify multiple modules by separating each module with comma. As an example:
> 
> ```text
> without extensions magento/adobe-stock-integration
> ```

Reviewer: ```suggestion
For `extensionRepo`, specify one or more extension repositories to remove before compiling your instance.  If you specify multiple repositories, use a comma after each repository. For example:

```text
without extensions magento/adobe-stock-integration
```
```
